nomt/src/beatree/branch/node.rs

```diff
@@ -294,11 +387,305 @@ impl BranchNodeBuilder {
         self.index += 1;
     }
 
+    pub fn push_chunk(&mut self, base: &BranchNode, from: usize, to: usize) {
```

Comment:
Looking at this function, it looks to me like it internalizes too much complexity, especially around handling edge cases.

the contract of this function is not clear. the typical case I expect we would optimize for is basically just a single bitwise memcpy of the separators and a normal memcpy of the node pointers, which is what we get when the prefix hasn't changed.

I don't see a strong case that this would be more efficient than calling `push` repeatedly in the edge cases, and we've also pessimized the base case with a bunch of branches and additional logic.

It seems better to have alternative variants for each expected code path.

My suggestion:
  - Have `push_chunk` variants for all changes in the prefix (none, shrink, grow). Our base case is none.
  - Have `push_chunk` variant for uncompressed entries.
  - Have the branch updater figure out which one to call in which situations. Figuring out which to call seems pretty easy during `BranchUpdater::build_branch`.

When the prefix is unchanged, we just need a single bitwise memcpy. When the prefix shrinks or grows, we need to do bitwise memcpy in a loop, but it should still be fast. Copying an uncompressed chunk would also be a single bitwise memcpy.

I think this should simplify and speed up the code substantially.